syncapi/sync/requestpool.go

```diff
@@ -132,10 +133,24 @@ func (rp *RequestPool) OnIncomingSyncRequest(req *http.Request, device *authtype
 
 func (rp *RequestPool) currentSyncForUser(req syncRequest, latestPos types.PaginationToken) (res *types.Response, err error) {
 	// TODO: handle ignored users
+	localpart, _, err := gomatrixserverlib.SplitID('@', req.device.UserID)
+	if err != nil {
+		return
+	}
+	var ignoredUserList []string
+	if data, e := rp.accountDB.GetAccountDataByType(req.ctx, localpart, "", "m.ignored_user_list"); e == nil && data != nil {
```

Comment:
We should handle errors correctly here, instead of ignoring it. Also, let's name the error `err` for consistency and clarity. What I'd suggest would be:

```go
data, err := rp.accountDB.GetAccountDataByType(req.ctx, localpart, "", "m.ignored_user_list")
if err != nil {
    return
}
if data != nil {
    // ...
}
```